source/agora/test/GossipProtocol.d

```diff
@@ -37,34 +48,58 @@ unittest
         assert(block_height == 0, block_height.to!string);
     }
 
-    Transaction tx =
-    {
-        [Input(hashFull("Message No. 1"))],
-        [Output(100)]
-    };
+    // It is validated. (the sum of `Output` <= the sum of `Input`)
+    // The Genesis block has 40,000,000
 
-    Hash tx_hash = hashFull(tx);
-    auto node_1 = network.apis.values[0];
-    node_1.putTransaction(tx);
+    // Creates the first transaction.
+    Transaction tx1 = Transaction(
+        [
+            Input(last_tx_hash, 0)
+        ],
+        [
+            Output(40_000_000, genesis_key_pare.address)
+        ]
+    );
+    Hash tx1Hash = hashFull(tx1);
+    tx1.inputs[0].signature = genesis_key_pare.secret.sign(tx1Hash[]);
+
+    node_1.putTransaction(tx1);
+    last_tx_hash = tx1Hash;
 
     // Check hasTransactionHash
     foreach (key, ref node; network.apis)
     {
-        assert(node.hasTransactionHash(tx_hash));
+        //  every nodes received tx
+        assert(node.hasTransactionHash(tx1Hash));
 
         auto block_height = node.getBlockHeight();
         assert(block_height == 1, block_height.to!string);
     }
 
-    tx.inputs[0].previous = hashFull("Message No. 2");
-    tx_hash = hashFull(tx);
-    node_1.putTransaction(tx);
+    // It isn't validated. (the sum of `Output` > the sum of `Input`)
```

Comment:
This should be a separate third transaction after two valid ones. Because otherwise the `getBlockHeight` API is not properly tested below.

Reply:
Modified~